hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java

```diff
@@ -449,13 +453,21 @@ public long optimizeDirDeletesAndSubmitRequest(long remainNum,
       deletedDirsCount.addAndGet(dirNum + subdirDelNum);
       movedDirsCount.addAndGet(subDirNum - subdirDelNum);
       movedFilesCount.addAndGet(subFileNum);
+      long timeTakenInIteration = Time.monotonicNow() - startTime;
       LOG.info("Number of dirs deleted: {}, Number of sub-dir " +
               "deleted: {}, Number of sub-files moved:" +
               " {} to DeletedTable, Number of sub-dirs moved {} to " +
-              "DeletedDirectoryTable, iteration elapsed: {}ms," +
+              "DeletedDirectoryTable, limit per iteration: {}, iteration elapsed: {}ms, " +
               " totalRunCount: {}",
-          dirNum, subdirDelNum, subFileNum, (subDirNum - subdirDelNum),
-          Time.monotonicNow() - startTime, getRunCount());
+          dirNum, subdirDelNum, subFileNum, (subDirNum - subdirDelNum), limit,
+          timeTakenInIteration, getRunCount());
+      if (remainNum <= 0) {
+        LOG.warn("Limit for no. of directory+files that can be deleted in one iteration is reached. " +
+                "Current limit: {} = {}",
+            OMConfigKeys.OZONE_PATH_DELETING_LIMIT_PER_TASK, ozoneManager.getConfiguration()
+                .getInt(OMConfigKeys.OZONE_PATH_DELETING_LIMIT_PER_TASK,
+                    OMConfigKeys.OZONE_PATH_DELETING_LIMIT_PER_TASK_DEFAULT));
+      }
```

Comment:
This shouldn't be a warn log, we can add it in a debug log and simplify it,

> if (remainNum <= 0 && LOG.isDebugEnabled()) {
      LOG.debug(
          "Limit for number of directories and files that can be processed in a deletion interval is reached. Total files and dirs moved: {}",
          (subFileNum + subDirNum - subdirDelNum));
    }

Reply:
I guess the Jira description doesn't currently specify what do log when the deletion limits are hit:
> At the end of each run, and INFO log should display how many items were processed, what the current limit of items per run is, and how long it took.
If the run time of the iteration exceeded the configured interval, the message should highlight this and be elevated to WARN.
This should also log the config keys that can remedy this for batch size, interval, or number of threads.

However based on past experiences troubleshooting slow deletion issues in a busy cluster with many other events going on I would be in favor of also elevating this to warn if the limits are being hit to indicate that the system is under load and deletion may be under-provisioned. However from the Jira this should be one single message as a summary with all this information at the corresponding log level based on its content.

Also worth noting that once the multi-threaded implementation is complete it should take a lot more to hit the deletion limit, and this message will not seem so spurious if 10 threads are being maxed out trying to process current deletion.

Reply:
However, with a million directories, the log could become flooded with this warning, but we can keep it for now.

Reply:
The number of log lines won't change, it will be one per iteration regardless of the conditions. We should definitely warn if it is taking longer than the interval. As for the key count limits being hit, it could go either way. One option is we could leave that condition at info for now, and as a follow up we could optionally save the state of the last few runs and only log a warning if the last 10 or so hit the key limits.

Reply:
Yes, that's a nice idea. We can warn if we hit the key limit more than 10 times.